File: app/src/main/java/com/turlir/abakcalc/ValueVisualizer.java
package com.turlir.abakcalc;

import com.turlir.calculator.converter.Printer;
import com.turlir.calculator.converter.Visual;

public class ValueVisualizer implements CalculatorVisual {

    private final Visual mDelegate;

    ValueVisualizer(Visual delegate) {
        mDelegate = delegate;
    }

    @Override
    public void print(Printer chain) {
        mDelegate.print(chain);
    }

    @Override
    public boolean selectionConstraint(int selStart, int selEnd, int length) {
        return false;
    }

    @Override
    public int[] interceptSelection(int a, int b) {
        return new int[0];
    }

    @Override
    public int constraintStart() {
        return 0;
    }

    @Override
    public int constraintEnd() {
        return 0;
    }

    @Override
    public String toString() {
        return mDelegate.toString();
    }
}

File: app/src/main/kotlin/com/turlir/abakcalc/ValueVisualizer.kt
package com.turlir.abakcalc

import com.turlir.calculator.converter.Printer
import com.turlir.calculator.converter.Visual

class ValueVisualizer constructor(private val mDelegate: Visual) : CalculatorVisual {

    override fun print(chain: Printer) {
        mDelegate.print(chain)
    }

    override fun selectionConstraint(selStart: Int, selEnd: Int, length: Int) = false

    override fun interceptSelection(nowS: Int, nowE: Int) = IntArray(0) // []

    override fun constraintStart() = 0

    override fun constraintEnd() = 0

    override fun toString() = mDelegate.toString()
}
